samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManagerFactory.scala

```diff
@@ -48,9 +49,24 @@ object KafkaCheckpointManagerFactory {
     } else {
       new KafkaConfig(config).getCheckpointSegmentBytes()
     }
-    (new Properties /: Map(
-      "cleanup.policy" -> "compact",
-      "segment.bytes" -> String.valueOf(segmentBytes))) { case (props, (k, v)) => props.put(k, v); props }
+
+    val isStreamMode = if (config != null) {
```

Comment:
since,`STREAM` is the default anyways, can we rewrite this to be more concise?

val isStreamMode = appConfig.getAppMode == ApplicationMode.STREAM